playbooks/provisioning/openstack/openstack_dns_records.yml

```diff
@@ -4,6 +4,11 @@
     private_records: "{{ private_records | default([]) + [ { 'type': 'A', 'hostname': hostvars[item]['ansible_hostname'], 'ip': hostvars[item]['private_v4'] } ] }}"
   with_items: "{{ groups['cluster_hosts'] }}"
 
+- name: "Add wildcard records to the private A records for infrahosts"
+  set_fact:
+    private_records: "{{ private_records | default([]) + [ { 'type': 'A', 'hostname': '*.' + openshift_app_domain, 'ip': hostvars[item]['openstack']['private_v4'] } ] }}"
```

Comment:
@day4skiing there's a slight difference between the CASL and the contrib repo - for this PR you need to remove the `[openstack]` portion as the implementation in contrib isn't "openstack aware" at this level.